src/filtrering/filtrering-filter/filterform/geografiskbosted-filterform.tsx

```diff
@@ -0,0 +1,139 @@
+import {FiltervalgModell} from '../../../model-interfaces';
+import * as React from 'react';
+import {useEffect, useState} from 'react';
+import classNames from 'classnames';
+import Grid from '../../../components/grid/grid';
+import {Checkbox, CheckboxGroup} from '@navikt/ds-react';
+import NullstillKnapp from '../../../components/nullstill-valg-knapp/nullstill-knapp';
+import {useGeografiskbostedSelector} from '../../../hooks/redux/use-geografiskbosted-selector';
+import {GeografiskBostedOptions} from '../../../ducks/geografiskBosted';
+import {MultiSelect} from 'react-multi-select-component';
+
+interface GeografiskBostedFilterformProps {
+    endreFiltervalg: (form: string, filterVerdi: string[]) => void;
+    filtervalg: FiltervalgModell;
+    gridColumns?: number;
+    emptyCheckboxFilterFormMessage?: string;
+}
+
+function GeografiskBostedFilterform({endreFiltervalg, filtervalg, gridColumns = 1}: GeografiskBostedFilterformProps) {
+    const [visGeografiskBosted, setVisGeografiskBosted] = useState<string[]>(filtervalg.visGeografiskBosted);
+    const [selectedGeografiskBosted, setSelectedGeografiskBosted] = useState<GeografiskBostedOptions[]>([]);
+    const [geografiskBostedOptions, setGeografiskBostedOptions] = useState<GeografiskBostedOptions[]>([]);
+    const geografiskbostedListData = useGeografiskbostedSelector();
+
+    useEffect(() => {
+        setVisGeografiskBosted(filtervalg.visGeografiskBosted);
+
+        if (geografiskbostedListData != null && geografiskbostedListData.size > 0) {
+            let selectedValues: GeografiskBostedOptions[] = [];
+            filtervalg.geografiskBosted.forEach(x => {
+                if (geografiskbostedListData.get(x) != null && geografiskbostedListData.get(x) !== undefined) {
```

Comment:
I think it would be possible to make this more readable, that is avoiding the explicit null-check and non-null assertion operator (which I personally try to avoid as much as possible, just like with the double-bang operator in Kotlin). In order to do so I would probably try to convert the Map to an Array of objects instead, which we could map-over (`[...].map(...)`) and filter using the `isDefined` custom type-guard (provided in the `/utils/types/typeguards.ts` file). This would probably result in the code being less performant though since `Map` has O(1)-lookup in most JavaScript-engines, but overall i don't think it would make much of a diference.

I tried to come up with a suggestion, but I didn't like the outcome (probably because I tried to do it quick and it became a little messy; a couple of iterations on the solution would probably do the trick).

If you want we can try to come up with a re-write together, but I also understand if you do not want this feature to be postponed.

Reply:
@slovrid I agree its not very readable, what you think about:

```
if (!isEmptyArray(geografiskbostedListData.size)) {
            let selectedValues: GeografiskBostedOptions[] = [];
            filtervalg.geografiskBosted
                .filter(bostedKode => geografiskbostedListData.get(bostedKode) != null)
                .filter(bostedKode => geografiskbostedListData.get(bostedKode) !== undefined)
                .forEach(bostedKode => {
                    selectedValues.push({
                        label: geografiskbostedListData.get(bostedKode)!,
                        value: bostedKode,
                        checked: true
                    });
                });
            setSelectedGeografiskBosted(selectedValues);
        }
```